pkg/packagekit/kpatch.jsx

```diff
@@ -0,0 +1,325 @@
+    }
+
+    installPatches(auto) {
+        return cockpit.spawn(["dnf", "-y", "kpatch", auto ? "auto" : "manual"], { superuser: "require", err: "message" });
+    }
+
+    checkSetup() {
+        return new Promise((resolve, reject) => {
+            cockpit.file("/etc/dnf/plugins/kpatch.conf").read()
+                    .then(data => {
+                        if (!data) {
+                            resolve();
+                        } else {
+                            const autoupdate = data.split("\n").find(l => l.indexOf("autoupdate") >= 0);
+                            const auto = autoupdate.endsWith("True");
+                            this.setState({ auto: auto, futureAuto: auto }, resolve);
+                        }
+                    })
+                    .catch(reject);
+        });
+    }
+
+    handleInstall() {
+        this.setState({ updating: true });
+        install_dialog(this.state.missing)
+                .then(() => {
+                    this.setState({ missing: [], updating: false });
+                })
+                .catch(() => this.setState({ updating: false }));
+    }
+
+    handleChange() {
+        this.setState({ updating: true });
+
+        const promises = [];
+
+        if (this.state.auto !== this.state.futureAuto)
+            promises.push(this.installPatches(this.state.futureAuto));
+
+        if (this.state.enabled !== this.state.futureEnabled)
+            promises.push(this.handleService(this.state.futureEnabled));
+
+        Promise.all(promises)
+                .then(() => this.setState({ showModal: false }))
+                .catch(console.log)
+                .finally(() => {
+                    this.checkSetup();
+                    this.setState({ updating: false });
+                });
+    }
+
+    render() {
+        // Not yet recognized
+        if (this.state.loaded === false)
+            return null;
+
+        // Not supported on this system
+        if (this.state.unavailable.length > 0)
+            return null;
+
+        let state = _("Enabled");
+        let actionText = _("Edit");
+        let action = () => this.setState({ showModal: true });
+
+        if (this.state.missing.length > 0 && this.state.unavailable.length === 0) {
+            state = _("Not installed");
+            actionText = _("Install");
+            action = this.handleInstall;
+        } else if (this.state.auto === false) {
+            state = _("Disabled");
+            actionText = _("Enable");
+        }
+
+        const details = [];
+        if (this.state.enabled !== this.state.futureEnabled) {
+            if (this.state.futureEnabled)
+                details.push(_("kpatch.service will be enabled. After each reboot patches will be applied."));
+            else
+                details.push(_("kpatch.service will be disabled. After reboot no patches will be applied."));
+        }
+        // FIXME - should we uninstall current patches as well?
```

Comment:
Should we?

Reply:
I'm inclined to say that this should not do any magic beyond `pkcon update`/`dnf update`. Cleanup of old kernels (and corresponding kpatches) belongs to the packaging system, not cockpit. It's IMHO not acceptable to pile up either of them with automatic updates or not paying attention.

Reply:
Agree.

Reply:
There's a middle ground: We could add a note in the settings modal that the package is still installed, but not used.

Reply:
There's a middle ground: We could add text to the modal about the package being installed, but unused.